Streetcode/Streetcode.WebApi/Extensions/WebApplicationExtensions.cs

```diff
@@ -47,4 +47,46 @@ public static class WebApplicationExtensions
             logger.LogError(ex, "An error occured during startup migration");
         }
     }
+
+    public static async Task ApplyMigrations(
+        this WebApplication app,
+        string scriptsFolderPath = "./Streetcode.DAL/Persistence/ScriptsMigration/")
+    {
+        using var scope = app.Services.CreateScope();
+        var logger = app.Services.GetRequiredService<ILogger<Program>>();
+        try
+        {
+            var streetcodeContext = scope.ServiceProvider.GetRequiredService<StreetcodeDbContext>();
+
+            IDbContextTransaction transaction = null;
+            try
+            {
+                var projRootDirectory = Directory.GetParent(Environment.CurrentDirectory)?.FullName!;
+
+                var scriptFiles = Directory.GetFiles($"{projRootDirectory}/{scriptsFolderPath}");
+
+                var filesContexts = await Task.WhenAll(scriptFiles.Select(file => File.ReadAllTextAsync(file)));
+                transaction = streetcodeContext.Database.BeginTransaction();
+
+                foreach (var task in filesContexts)
```

Comment:
Perhaps, it's a fileContext or some single sqlScript, but not a task

Reply:
Agree